server/src/router/v1/index.ts

```diff
@@ -853,13 +854,218 @@ v1App.openapi(
+				},
+			},
+			params: z.object({
+				major: z.string(),
+			}),
+		},
+		responses: {
+			[HttpStatusCodes.NO_CONTENT]: {
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
+			[HttpStatusCodes.CONFLICT]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Conflict',
+			},
+		},
+	}),
+	async (c) => {
+		const { name } = c.req.valid('json');
+		const { major } = c.req.valid('param');
+
+		const newMajor = await db
+			.update(majors)
+			.set({ name })
+			.where(eq(majors.name, major))
+			.returning();
+		if (newMajor.length === 0) {
+			return c.json({ error: 'Major not updated' }, HttpStatusCodes.CONFLICT);
+		}
+		return c.text('', HttpStatusCodes.NO_CONTENT);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'delete',
+		path: '/majors/{major}',
+		tags: ['majors'],
+		summary: 'delete major',
+		middleware: [authMiddleWare('admin')],
+		request: {
+			params: z.object({
+				major: z.string(),
+			}),
+		},
+		responses: {
+			[HttpStatusCodes.NO_CONTENT]: {
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Unauthorized',
+			},
+			[HttpStatusCodes.FORBIDDEN]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Forbidden',
+			},
+			[HttpStatusCodes.NOT_FOUND]: {
+				content: {
+					'application/json': {
+						schema: z.object({
+							error: z.string(),
+						}),
+					},
+				},
+				description: 'Conflict',
+			},
+		},
+	}),
+	async (c) => {
+		const { major } = c.req.valid('param');
+		
+		const newMajor = await db
+			.delete(majors)
+			.where(eq(majors.name, major))
+			.returning();
+
+		if (newMajor.length === 0) {
+			return c.json({ error: 'Major not deleted' }, HttpStatusCodes.NOT_FOUND);
+		}
+		return c.text('', HttpStatusCodes.NO_CONTENT);
+	},
+);
+
+v1App.openapi(
+	createRoute({
+		method: 'post',
+		path: '/majors',
+		tags: ['majors'],
+		summary: 'create major',
+		middleware: [authMiddleWare('admin')],
+		request: {
+			body: {
+				content: {
+					'application/json': {
+						schema: majorSchema,
+					},
+				},
+			},
+		},
+		responses: {
+			[HttpStatusCodes.CREATED]: {
+				content: {
+					'application/json': {
+						schema: majorSchema,
+					},
+				},
+				description: 'Successful response',
+			},
+			[HttpStatusCodes.UNAUTHORIZED]: {
```

Comment:
for unauthorized requests
we can use `...unauthorizedRequest` like what we did in other routes because this is a common one we can reuse